### PR TITLE
Test exclude group from receiving shares

### LINKED ISSUE #35671: Write sharing API test when a group is blacklisted from sharing
We don't have tests for blacklisting groups from receiving/sharing shares, and sharing using the api. 

We do have webUI tests for this scenario, so, please check that for possible scenarios. Part of #31531  

Grandparent issue owncloud/QA#353

### PR DESCRIPTION
**Pull request by individual-it**
## Description
test if groups that are blacklisted from receiving a share, cannot receive a share

## Related Issue

- Fixes #35671

## Motivation and Context
more tests, more better

## How Has This Been Tested?
:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 1]
```diff
@@ -1794,6 +1796,47 @@ public function assertThatNoSharesAreSharedWithUser($user) {
 		);
 	}
 
+	/**
+	 * @When the administrator adds group :group to the exclude group from sharing list
```
Comment (dpakach) [inline tests/acceptance/features/bootstrap/Sharing.php]:
```suggestion
	 * @When the administrator adds group :group to the exclude group from sharing list using the occ command
```

File: tests/acceptance/features/apiShareManagementBasic/excludeGroupFromReceivingShares.feature [hunk 2]
```diff
@@ -0,0 +1,135 @@
+@api @TestAlsoOnExternalUserBackend
+Feature: sharing
```
Comment (dpakach) [inline tests/acceptance/features/apiShareManagementBasic/excludeGroupFromReceivingShares.feature]: I think it should be described in more detail here

**Comment by individual-it** [comment]
@dpakach implemented the suggestions, please review again

**Comment by dpakach** [comment]
Changes LGTM but it is failing for some reason.

**Review by phil-davis — COMMENTED** [review]
I only found 1 little typo

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 3]
```diff
@@ -1794,6 +1796,41 @@ public function assertThatNoSharesAreSharedWithUser($user) {
...
 	}
 
+	/**
+	 * @When the administrator adds group :group to the exclude group from sharing list using the occ command
+	 *
+	 * @param string $group
+	 *
+	 * @return void
+	 */
+	public function adminstratorAddedGroupToExcludeFromSharingList($group) {
```
[the remark this answers is not in the document]
Comment (individual-it) [reply tests/acceptance/features/bootstrap/Sharing.php]: done, please review again

**Review by phil-davis — APPROVED** [review]

**Review by dpakach — APPROVED** [review]

**Comment by individual-it** [comment]
backport in #35778

**Pull request merged**